src/Eval.hs

```diff
@@ -65,17 +65,24 @@ eval ctx xobj@(XObj o info ty) preference =
     Lst body -> eval' body
     Sym spath@(SymPath p n) _ ->
       pure $
-        fromMaybe
-          (evalError ctx ("Can't find symbol '" ++ show n ++ "'") info) -- all else failed, error.
-          -- Certain contexts prefer looking up bindings in the dynamic environment (e.g. defdyanmic) while others
-          -- prefer the static global environment.
-          ( ( case preference of
+        case tryAllLookups of
+          Just (_, (Right (XObj (Lst ((XObj Def _ _) : _)) _ _))) ->
+            (ctx, Left (HasStaticCall xobj info))
+          Just (_, (Right (XObj (Lst ((XObj (Defn _) _ _) : _)) _ _))) ->
+            (ctx, Left (HasStaticCall xobj info))
+          Just (_, (Right (XObj (Lst ((XObj (External _) _ _) : _)) _ _))) ->
+            (ctx, Left (HasStaticCall xobj info))
```

Comment:
This is somewhat clunky, I’d be happy to hear alternatives.

Reply:
What about moving this into a function called `checkStatic` or `isStatic` and call it in the Maybe Monad? Then you can selectively call it for lookups that matter as well (unless it matters for all of them, in which case tack it at the end of the maybe chain instead)

```haskell
checkStatic :: (Context, Either EvalError XObj)-> Maybe (Context, Either EvalError XObj)
checkStatic ctx (Right XObj (Lst ((XObj Def ...)))) = Just (ctx, Left HasStaticCall....)
...
checkStatic ctx anything = pure (ctx, anything)

--- assuming it only applies for certain lookups:
(case preference of 
  Dyna -> tryDynamicLookup >>= checkStatic
  Global -> (tryLookup <|> tryDynamicLookup) >>= checkStatic
)

--- or just tack it to the end of the chain
(case preference of 
  Dyna -> tryDynamicLookup
  Global -> (tryLookup <|> tryDynamicLookup)
) <|> (if null p then tryInternalLookup spath else tryLookup spath)
  >>= checkStatic
```

This way it's a lot easier to pull this out or change it later.